crypto/modes/asm/ghashv8-armx.pl

```diff
@@ -70,6 +70,7 @@
 $code=<<___;
 #include "arm_arch.h"
 
+#if __ARM_MAX_ARCH__>=7
 .text
 ___
 $code.=".arch	armv8-a+crypto\n"	if ($flavour =~ /64/);
```

Comment:
why is there an `#undef  __thumb2__` a few lines below?

Reply:
I'm just asking myself if defining/un-defining a predefined macro is the right way to do.

Reply:
Well, there is nothing magic or "sacred" about predefined macros. They just tell the story of initial condition, but they are not prohibited to be undefined. As long as there is clear understanding of what does undefining one entail.

Reply:
Either way, I'm going to treat this remark as side note in the context of this request. It's not the only occurrence of this, so if there is an alternative to discuss and implement, this request is arguably not right spot.

Reply:
Yes, this is only a general remark.

Reply:
As for undefining predefines. I [think I] understand the reluctance, but there is one thing that is special when it comes to assembler. In compiler context pre-defines do reflect how code is generated, so that undefining them can be viewed as "strange". While in assembler case they have no bearing on code generation, in sense that assembler is still in actual control. For example even if you observe `__thumb2__` among predefines, default code generation mode is still non-Thumb, it has to be engaged explicitly with corresponding directive. Predefines in assembly preprocessing are really just reflection of command line options for programmer to consider (and dismiss if so desired ;-).